packages/ui/src/forms/Form/fieldReducer.ts

```diff
@@ -398,6 +398,15 @@ export function fieldReducer(state: FormState, action: FieldAction): FormState {
         [action.path]: newField,
       }
 
+      // reset `isModified` in all other fields
+      if ('value' in action) {
+        for (const [path, field] of Object.entries(newState)) {
+          if (path !== action.path && 'isModified' in field) {
+            delete newState[path].isModified
```

Comment:
Note: this is not ideal, but works for now. An alternate approach would be to store the `lastModifiedPath` on the form context itself. That would remove this loop. The problem is we don't have access to that context from within the field reducer.